### ISSUE TITLE
Support for more color formats

### ISSUE THREAD

**Issue by jonmmease** [pyviz/param#355]
Right now, the values of a `param.Color` parameter may only be specified in hex format.  In addition, it would be nice to allow:
  - css [named colors](https://www.w3schools.com/colors/colors_names.asp) (e.g. `"red"`, `"crimson"`, `"pink"`) 
  - rgb(a), hsl(a) (e.g. `"rgb(128, 128, 64)"`): See https://www.w3schools.com/cssref/css_colors_legal.asp

Do you foresee any downstream ecosystem problems associated with expanding the set of allowed colors?

**Comment by jbednar** [comment]
Are you anticipating converting the provided colors into hex when set, or passing them straight through? Converting all these would be a bit complicated and probably too much code for Param to have, so I assume straight through. But if passing straight through, existing user code with param.Color parameters may or may not be able to handle the other color formats, so maybe allowing them could be an explicit flag to set when the parameter is defined. In that case, sure!

**Comment by jonmmease** [comment]
Yes, I was picturing passing them straight through.  A flag parameter in the `param.Color` constructor sounds like a good idea. Maybe just `css=False` as the default, then setting this to `True` would allow any valid css color as a specification.

**Comment by philippjfr** [comment]
I'd also be fine with a `css` or ``allow_css` flag.

**Comment by philippjfr** [comment]
Main question I guess is where we'd get the list of valid named colors from?

**Comment by jonmmease** [comment]
> Main question I guess is where we'd get the list of valid named colors from?

Datashader and plotly.py already have a `dict`/`list` of the color literals with all of the valid namedcolors. So I was planning to copy one of those over.  I'll probably also grab the validation regular expressions from plotly.py as well.

**Comment by jbednar** [comment]
That sounds a bit dubious as a contribution to param; param doesn't seem like the place to be hard-coding a list of color names.  Maybe instead the constructor for this parameter type could have an argument allowing an extra list or dictionary of allowed names, and the validator would allow all the usual values plus whatever's on that list?  Otherwise, it seems very likely that whatever is hardcoded into Param won't actually match what's legal for a particular usage of the param.Color value with some external library. Not sure about the regexes; that may or may not make sense to hardcode (depending on what they allow).
